src/cf.data.pre

```diff
@@ -7112,11 +7112,28 @@ LOC: Config.visibleHostname
 DEFAULT: none
 DEFAULT_DOC: Automatically detect the system host name
 DOC_START
-	If you want to present a special hostname in error messages, etc,
-	define this.  Otherwise, the return value of gethostname()
-	will be used. If you have multiple caches in a cluster and
-	get errors about IP-forwarding you must set them to have individual
-	names with this setting.
+	The hostname of this proxy.
+
+	Hosts connected to the Internet or handling messages which travel
```

Comment:
This paragraph sets the context for behaviour described in the following paragraphs. Removing or placing it last makes all this seems like arbitrary decisions we are imposing on admin ourselves. Removing it also perpetuates the admin misbelief that they can safely set it to random strings.

Reply:
> The whole point of this directive is to change the name of the host where Squid runs as seen by external/remote/public entities.

I disagree. Those external/remote/public entities see a proxy service first and a host (hosting a proxy service) second (or not at all). In fact, in many contexts, when addressing the proxy service by the name configured here, those entities will not talk to the host where this Squid is running!

> This paragraph sets the context for behaviour described in the following paragraphs.

I am not sure it does. Naming is needed not because some RFC requires hosts to be named. Naming is needed for the use cases we describe further below. The first/previous paragraph is enough to set the general context IMO.

> Removing or placing it last makes all this seems like arbitrary decisions we are imposing on admin ourselves. 

I disagree. We document where this name is used. There is nothing "arbitrary" about those locations -- they are all driven by the same need to name "this proxy" or "this set of proxies"  for addressing it via HTTP and other relevant protocols. The admin should be able to make this naming decision based on the subset of the applicable-to-the-admin use cases (and the sited RFC would be of little, if any, help to that admin).